Assets/_System/Scripts/TurnBasedScripts/TileSelector.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TileSelector : MonoBehaviour
+{
+    public Vector2Int currentPos;
+    private GridTile targetTile;
+    private Vector2Int moveDir;
+    // Start is called before the first frame update
+    void Start()
+    {
+        moveDir = new Vector2Int(0, 0);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        GetInput();
+        if(moveDir != new Vector2Int(0,0))
+        {
+            MoveSelector();
+        }
+    }
+
+    private void MoveSelector()
+    {
+        targetTile = GridManager.GetGridTileAtPosition(currentPos + moveDir);
+
+        if(targetTile != null)
+        {
+            GridManager.Instance.UnsetHoveredTile(GridManager.GetGridTileAtPosition(currentPos));
+            currentPos += moveDir;
+            GridManager.Instance.SetHoveredTile(GridManager.GetGridTileAtPosition(currentPos));
+            transform.position = GridManager.GetGridTileAtPosition(currentPos).m_WorldPosition; 
+        }
+        moveDir = new Vector2Int(0,0);
+        targetTile = null;
+    }
+
+    private void GetInput()
+    {
+        if (Input.GetKeyDown(KeyCode.D) )
```

Comment:
Not a need to do now, but sometime soon we will want to switch this out the Input Service.